### PR TITLE
ai folder merge .tpp and .h to .hpp

### LINKED ISSUE #2311: Merge .tpp and .h files in the software/ai/ folder
### Description of the task
Since .tpp files don't play nicely in IDEs, we want to combine the .h and .tpp files into one .hpp file. The implementation portions of the .tpp can be at the bottom of the .hpp file

### Acceptance criteria

- [ ] as described for the software/ai/ folder

### PR DESCRIPTION
**Pull request by jameschuang2002**
### Description
  merge all .h and .tpp files in ai folder to .hpp

### Testing Done
   build file changed to .hpp and build completed successfully for all files

### Resolved Issues
resolves #2311 

### Length Justification
N/A

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**
Most items below are not relevant to this task.
- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/ai/passing/pass_evaluation.hpp [hunk 1]
```diff
@@ -95,5 +95,78 @@ class PassEvaluation
     // The timestamp when this evaluation was created
     Timestamp timestamp_;
 };
+#include "software/ai/passing/pass.h"
```
Comment (jonathanlew) [inline src/software/ai/passing/pass_evaluation.hpp]: please put this header at the top of the file

**Review by jonathanlew — CHANGES_REQUESTED** [review]
just one comment from me

**Review by akhilveeraghanta — COMMENTED** [review]
lgtm pending Jons comment

**Comment by Bvasilchikov** [comment]
lgtm to me as well

**Review by jonathanlew — APPROVED** [review]

**Review by Bvasilchikov — APPROVED** [review]
Looks good

**Review by naraiamsakun — APPROVED** [review]

**Pull request merged**